### DISCUSSION TITLE
ext::intel::info::device::free_memory

### DISCUSSION THREAD

**Discussion by jinz2014** [intel/llvm#13652] [Q&A] [answered]
Could you comment on the support of ext::intel::info::device::free_memory on Intel GPUs ?

**Comment by steffenlarsen** [comment]
The `ext::intel::info::device:free_memory` device query is supported on Intel GPUs, though they do currently require the user to set the [`ZES_ENABLE_SYSMAN` env var](https://spec.oneapi.io/level-zero/latest/sysman/PROG.html).

**Comment by AlexeySachkov** [comment, marked as answer]
Hi @jinz2014,

The API interface itself is device-agnostic, so it should work everywhere. However, implementation details of it in low-level parts of the SW stack do indeed introduce some differences.

For example, according to the [extension spec](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_intel_device_info.md#free-global-memory) the query may need `ZES_ENABLE_SYSMAN=1` env variable to be set for it to work. However, as far as I know this functionality is not supported on integrated GPUs, i.e. you will see 0 as a result regardless of env variables. Unfortunately, I don't have any links to a more precise documentation, or issue submitted about that behavior.
